uk_bin_collection/uk_bin_collection/councils/UttlesfordDistrictCouncil.py

```
@@ -63,7 +63,24 @@ def parse_data(self, page: str, **kwargs) -> dict:
             logging.info("Selecting address")
             drop_down_values = Select(input_element_postcode_dd)
 
-            drop_down_values.select_by_visible_text(str(user_paon))
+            # Try exact match first, fall back to house number prefix match
+            matched = False
+            try:
+                drop_down_values.select_by_visible_text(str(user_paon))
+                matched = True
+            except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)